### PR TITLE
Updates and fixes out of data AWS packer and terraform code

### PR DESCRIPTION
**Pull request by nickwales**
The "Provision a Nomad cluster in the cloud" works in AWS with these updates to
- use an available ubuntu version
- uses hashicorp packages where possible

### CODE REVIEW

File: terraform/shared/config/consul.hcl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+advertise_addr = "IP_ADDRESS"
+bind_addr = "0.0.0.0"
```
[the remark this answers is not in the document]
Comment (nickwales) [reply terraform/shared/config/consul.hcl:2]: Agreed, this demo environment is lacking the security we'd expect in production. We could follow up with some improvements in that regard.

File: terraform/shared/scripts/setup.sh [hunk 2]
```diff
@@ -3,173 +3,85 @@
...
+sudo apt-get install -yq  apt-utils
 
-CONSULVERSION=1.12.2
-CONSULDOWNLOAD=https://releases.hashicorp.com/consul/${CONSULVERSION}/consul_${CONSULVERSION}_linux_amd64.zip
-CONSULCONFIGDIR=/etc/consul.d
-CONSULDIR=/opt/consul
+# Install HashiCorp products
+CONSULVERSION=1.17.0
+VAULTVERSION=1.15.4
+NOMADVERSION=1.7.1
```
[the remark this answers is not in the document]
Comment (nickwales) [reply terraform/shared/scripts/setup.sh]: Done!

File: terraform/shared/scripts/client.sh [hunk 3]
```diff
@@ -26,34 +24,40 @@ if [ "$CLOUD" = "gce" ]; then
 else
   IP_ADDRESS=$(curl http://instance-data/latest/meta-data/local-ipv4)
 fi
-# IP_ADDRESS="$(/sbin/ifconfig eth0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
+
+# Systemd-resolved config
```
[the remark this answers is not in the document]
Comment (nickwales) [reply terraform/shared/scripts/client.sh]: Explained and added link to the documentation

File: terraform/shared/scripts/setup.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikaelHashiCorp) [reply terraform/shared/scripts/setup.sh]: I suggest to bump all versions to latest point release before merge.

[the remark this answers is not in the document]
Comment (nickwales) [reply terraform/shared/scripts/setup.sh]: Done

**Review by tgross — CHANGES_REQUESTED** [review]
@nickwales can you run `terraform fmt` on the Terraform files here? `make check` is failing:

```
==> Formatting HCL
make[1]: Leaving directory '/home/runner/work/nomad/nomad'
the following HCL files are out of sync
 M terraform/aws/modules/hashistack/hashistack.tf
 M terraform/shared/config/consul.hcl
 M terraform/shared/config/consul_client.hcl
```

Once that's done, this should be good to merge. I'd really love it if we could circle back and make this more production-ready with recommended configuration for security, but we can do that under a separate PR.

**Review by tgross — APPROVED** [review]

**Pull request merged**